tsl/src/compression/compress_utils.c

```diff
@@ -208,6 +209,66 @@ compresschunkcxt_init(CompressChunkCxt *cxt, Cache *hcache, Oid hypertable_relid
 	return;
 }
 
+static void
+preserve_uncompressed_chunk_stats(Oid chunk_relid)
+{
+	AlterTableCmd at_cmd = {
+		.type = T_AlterTableCmd,
+		.subtype = AT_SetRelOptions,
+		.def = (Node *) list_make1(
+			makeDefElem("autovacuum_enabled", (Node *) makeString("false"), -1)),
+	};
+	VacuumRelation vr = {
+		.type = T_VacuumRelation,
+		.relation = NULL,
+		.oid = chunk_relid,
+		.va_cols = NIL,
+	};
+	VacuumStmt vs = {
+		.type = T_VacuumStmt,
+		.rels = list_make1(&vr),
+#if PG12_GE
+		.is_vacuumcmd = false,
+		.options = NIL,
+#else
+		.options = VACOPT_ANALYZE,
+#endif
+	};
+
+#if PG12_GE
+	ExecVacuum(NULL, &vs, true);
+#else
+	ExecVacuum(&vs, true);
+#endif
+	AlterTableInternal(chunk_relid, list_make1(&at_cmd), false);
+}
+
+/* This function is intended to undo the disabling of autovacuum done when we compressed a chunk.
+ * Note that we do not cache the previous value for this (as we don't expect users to toggle this
```

Comment:
I guess this is OK, but just want to point out the this will potentially revert changes that someone has done to the vacuum setting on an individual chunk.